File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/PartyRelationshipDocumentation.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class PartyRelationshipDocumentation : ISerialized
    {
        public PartyRelationshipDocumentation(XmlNode xmlNode)
        {
            XmlNode masterAgreementNode = xmlNode.SelectSingleNode("masterAgreement");
            
            if (masterAgreementNode != null)
            {
                if (masterAgreementNode.Attributes["href"] != null || masterAgreementNode.Attributes["id"] != null) 
                {
                    if (masterAgreementNode.Attributes["id"] != null) 
                    {
                        masterAgreementIDRef_ = masterAgreementNode.Attributes["id"].Value;
                        MasterAgreement ob = new MasterAgreement(masterAgreementNode);
                        IDManager.SetID(masterAgreementIDRef_, ob);
                    }
                    else if (masterAgreementNode.Attributes["href"] != null)
                    {
                        masterAgreementIDRef_ = masterAgreementNode.Attributes["href"].Value;
                    }
                    else
                    {
                        masterAgreement_ = new MasterAgreement(masterAgreementNode);
                    }
                }
                else
                {
                    masterAgreement_ = new MasterAgreement(masterAgreementNode);
                }
            }
            
        
            XmlNode creditSupportAgreementNode = xmlNode.SelectSingleNode("creditSupportAgreement");
            
            if (creditSupportAgreementNode != null)
            {
                if (creditSupportAgreementNode.Attributes["href"] != null || creditSupportAgreementNode.Attributes["id"] != null) 
                {
                    if (creditSupportAgreementNode.Attributes["id"] != null) 
                    {
                        creditSupportAgreementIDRef_ = creditSupportAgreementNode.Attributes["id"].Value;
                        CreditSupportAgreement ob = new CreditSupportAgreement(creditSupportAgreementNode);
                        IDManager.SetID(creditSupportAgreementIDRef_, ob);
                    }
                    else if (creditSupportAgreementNode.Attributes["href"] != null)
                    {
                        creditSupportAgreementIDRef_ = creditSupportAgreementNode.Attributes["href"].Value;
                    }
                    else
                    {
                        creditSupportAgreement_ = new CreditSupportAgreement(creditSupportAgreementNode);
                    }
                }
                else
                {
                    creditSupportAgreement_ = new CreditSupportAgreement(creditSupportAgreementNode);
                }
            }
            
        
            XmlNode agreementNode = xmlNode.SelectSingleNode("agreement");
            
            if (agreementNode != null)
            {
                if (agreementNode.Attributes["href"] != null || agreementNode.Attributes["id"] != null) 
                {
                    if (agreementNode.Attributes["id"] != null) 
                    {
                        agreementIDRef_ = agreementNode.Attributes["id"].Value;
                        GenericAgreement ob = new GenericAgreement(agreementNode);
                        IDManager.SetID(agreementIDRef_, ob);
                    }
                    else if (agreementNode.Attributes["href"] != null)
                    {
                        agreementIDRef_ = agreementNode.Attributes["href"].Value;
                    }
                    else
                    {
                        agreement_ = new GenericAgreement(agreementNode);
                    }
                }
                else
                {
                    agreement_ = new GenericAgreement(agreementNode);
                }
            }
            
        
        }
        
    
        #region MasterAgreement_
        private MasterAgreement masterAgreement_;
        public MasterAgreement MasterAgreement_
        {
            get
            {
                if (this.masterAgreement_ != null)
                {
                    return this.masterAgreement_; 
                }
                else if (this.masterAgreementIDRef_ != null)
                {
                    masterAgreement_ = IDManager.getID(masterAgreementIDRef_) as MasterAgreement;
                    return this.masterAgreement_; 
                }
                else
                {
                      return this.masterAgreement_; 
                }
            }
            set
            {
                if (this.masterAgreement_ != value)
                {
                    this.masterAgreement_ = value;
                }
            }
        }
        #endregion
        
        public string masterAgreementIDRef_ { get; set; }
        #region CreditSupportAgreement_
        private CreditSupportAgreement creditSupportAgreement_;
        public CreditSupportAgreement CreditSupportAgreement_
        {
            get
            {
                if (this.creditSupportAgreement_ != null)
                {
                    return this.creditSupportAgreement_; 
                }
                else if (this.creditSupportAgreementIDRef_ != null)
                {
                    creditSupportAgreement_ = IDManager.getID(creditSupportAgreementIDRef_) as CreditSupportAgreement;
                    return this.creditSupportAgreement_; 
                }
                else
                {
                      return this.creditSupportAgreement_; 
                }
            }
            set
            {
                if (this.creditSupportAgreement_ != value)
                {
                    this.creditSupportAgreement_ = value;
                }
            }
        }
        #endregion
        
        public string creditSupportAgreementIDRef_ { get; set; }
        #region Agreement_
        private GenericAgreement agreement_;
        public GenericAgreement Agreement_
        {
            get
            {
                if (this.agreement_ != null)
                {
                    return this.agreement_; 
                }
                else if (this.agreementIDRef_ != null)
                {
                    agreement_ = IDManager.getID(agreementIDRef_) as GenericAgreement;
                    return this.agreement_; 
                }
                else
                {
                      return this.agreement_; 
                }
            }
            set
            {
                if (this.agreement_ != value)
                {
                    this.agreement_ = value;
                }
            }
        }
        #endregion
        
        public string agreementIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/PartyRelationshipDocumentation.cpp
// PartyRelationshipDocumentation.cpp 
#include "PartyRelationshipDocumentation.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

PartyRelationshipDocumentation::PartyRelationshipDocumentation(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //masterAgreementNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* masterAgreementNode = xmlNode->FirstChildElement("masterAgreement");

   if(masterAgreementNode){masterAgreementIsNull_ = false;}
   else{masterAgreementIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- masterAgreementNode , address : " << masterAgreementNode << std::endl;
   #endif
   if(masterAgreementNode)
   {
      if(masterAgreementNode->Attribute("href") || masterAgreementNode->Attribute("id"))
      {
          if(masterAgreementNode->Attribute("id"))
          {
             masterAgreementIDRef_ = masterAgreementNode->Attribute("id");
             masterAgreement_ = boost::shared_ptr<MasterAgreement>(new MasterAgreement(masterAgreementNode));
             masterAgreement_->setID(masterAgreementIDRef_);
             IDManager::instance().setID(masterAgreementIDRef_,masterAgreement_);
          }
          else if(masterAgreementNode->Attribute("href")) { masterAgreementIDRef_ = masterAgreementNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { masterAgreement_ = boost::shared_ptr<MasterAgreement>(new MasterAgreement(masterAgreementNode));}
   }

   //creditSupportAgreementNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* creditSupportAgreementNode = xmlNode->FirstChildElement("creditSupportAgreement");

   if(creditSupportAgreementNode){creditSupportAgreementIsNull_ = false;}
   else{creditSupportAgreementIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- creditSupportAgreementNode , address : " << creditSupportAgreementNode << std::endl;
   #endif
   if(creditSupportAgreementNode)
   {
      if(creditSupportAgreementNode->Attribute("href") || creditSupportAgreementNode->Attribute("id"))
      {
          if(creditSupportAgreementNode->Attribute("id"))
          {
             creditSupportAgreementIDRef_ = creditSupportAgreementNode->Attribute("id");
             creditSupportAgreement_ = boost::shared_ptr<CreditSupportAgreement>(new CreditSupportAgreement(creditSupportAgreementNode));
             creditSupportAgreement_->setID(creditSupportAgreementIDRef_);
             IDManager::instance().setID(creditSupportAgreementIDRef_,creditSupportAgreement_);
          }
          else if(creditSupportAgreementNode->Attribute("href")) { creditSupportAgreementIDRef_ = creditSupportAgreementNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { creditSupportAgreement_ = boost::shared_ptr<CreditSupportAgreement>(new CreditSupportAgreement(creditSupportAgreementNode));}
   }

   //agreementNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* agreementNode = xmlNode->FirstChildElement("agreement");

   if(agreementNode){agreementIsNull_ = false;}
   else{agreementIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- agreementNode , address : " << agreementNode << std::endl;
   #endif
   if(agreementNode)
   {
      if(agreementNode->Attribute("href") || agreementNode->Attribute("id"))
      {
          if(agreementNode->Attribute("id"))
          {
             agreementIDRef_ = agreementNode->Attribute("id");
             agreement_ = boost::shared_ptr<GenericAgreement>(new GenericAgreement(agreementNode));
             agreement_->setID(agreementIDRef_);
             IDManager::instance().setID(agreementIDRef_,agreement_);
          }
          else if(agreementNode->Attribute("href")) { agreementIDRef_ = agreementNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { agreement_ = boost::shared_ptr<GenericAgreement>(new GenericAgreement(agreementNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<MasterAgreement> PartyRelationshipDocumentation::getMasterAgreement()
{
   if(!this->masterAgreementIsNull_){
        if(masterAgreementIDRef_ != ""){
             return boost::shared_static_cast<MasterAgreement>(IDManager::instance().getID(masterAgreementIDRef_));
        }else{
             return this->masterAgreement_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<MasterAgreement>();
   }
}
boost::shared_ptr<CreditSupportAgreement> PartyRelationshipDocumentation::getCreditSupportAgreement()
{
   if(!this->creditSupportAgreementIsNull_){
        if(creditSupportAgreementIDRef_ != ""){
             return boost::shared_static_cast<CreditSupportAgreement>(IDManager::instance().getID(creditSupportAgreementIDRef_));
        }else{
             return this->creditSupportAgreement_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CreditSupportAgreement>();
   }
}
boost::shared_ptr<GenericAgreement> PartyRelationshipDocumentation::getAgreement()
{
   if(!this->agreementIsNull_){
        if(agreementIDRef_ != ""){
             return boost::shared_static_cast<GenericAgreement>(IDManager::instance().getID(agreementIDRef_));
        }else{
             return this->agreement_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<GenericAgreement>();
   }
}
}

